### ISSUE TITLE
Understanding dropouts

### ISSUE THREAD

**Issue by ijmbarr** [fchollet/keras#3691]
I'm trying to understand how dropouts are implemented within Keras. 

When are units to block in the dropouts layer set? Can they be set to specific configurations? Are they reset everytime I call train_on_batch?, or is there a way to get the same dropout configuration to persist over multiple calls, or is a different one created each time?

Thanks

**Comment by anjishnu** [comment]
I'm curious too, a pointer to which parts of the code to look at to understand the dropout implementation would be really helpful.

**Comment by anjishnu** [comment]
I think this piece of code is instructive - https://github.com/fchollet/keras/blob/master/keras/layers/recurrent.py

The swapping of tensors between training and test is happening in the backend here: 
https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py

I guess one thing I don't quite get is that they are are scaling up the masked output during the forward pass instead of scaling down the vector in the backprop - I am not 100% sure if the two are completely equivalent.
